helm/polaris/templates/deployment.yaml

```diff
@@ -0,0 +1,119 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "polaris.fullname" . }}
+  labels:
+    {{- include "polaris.labels" . | nindent 4 }}
+    {{- if .Values.podLabels }}
+    {{- tpl (toYaml .Values.podLabels) . | nindent 4 }}
+    {{- end }}
+spec:
+  {{- if not .Values.autoscaling.enabled }}
+  replicas: {{ .Values.replicaCount }}
+  {{- end }}
+  selector:
+    matchLabels:
+      {{- include "polaris.selectorLabels" . | nindent 6 }}
+  template:
+    metadata:
+      annotations:
+        {{- if .Values.podAnnotations }}
+          {{- tpl (toYaml .Values.podAnnotations) . | nindent 8 }}
+        {{- end }}
+      labels:
+        {{- include "polaris.selectorLabels" . | nindent 8 }}
+        {{- if .Values.podLabels }}
+        {{- tpl (toYaml .Values.podLabels) . | nindent 8 }}
+        {{- end }}
+    spec:
+      {{- if .Values.persistenceConfigSecret }}
+      initContainers:
+        - name: init-config
+          image: "{{ tpl .Values.initImage.repository . }}:{{ tpl .Values.initImage.tag . }}"
+          imagePullPolicy: {{ tpl .Values.initImage.pullPolicy . }}
+          command: ['sh', '-c', 'cd /secret && jar -cf /tmp/conf.jar persistence.xml']
+          volumeMounts:
+            - name: tmp-volume
+              mountPath: /tmp
```

Comment:
IMHO, it would be nice to avoid mounting as `/tmp`. It's a standard location for temporary files, so mounting an external volume there looks suspicious. Plus, it looks odd that a file we actually care to save and reuse (I assume) is under `/tmp`. By the same logic, the name should probably be something like `conf-volume`. WDYT?

Reply:
This path is derived from https://github.com/polaris-catalog/polaris/pull/88. Basically, we need a shared volume between init container and actual runtime container. The init container will then be responsible of generating the config-jar. If we can agreed on a path, I can make the needed change. We can't use `conf-volume` as that is already being used for the `polaris-service.yaml`:
```
          volumeMounts:
            - name: config-volume
              mountPath: /app/config/polaris-server.yml
              subPath: polaris-server.yml
            {{- if .Values.persistenceConfigSecret }}
            - name: tmp-volume
              mountPath: /tmp
            {{- end }}
...
      volumes:
        - name: config-volume
          configMap:
            name: {{ include "polaris.fullname" . }}
```

Reply:
Maybe called it db-config-volume then have it mounts on `/app/db-config`?

Reply:
#88 does not require the config jar to be in `/tmp`. I think any directory name should work (suggestion: `/eclipselink-config`).

Also, in #88 it looks like we still have to set a server config property to point to e.g. `/eclipselink-config/conf.jar`... I do not see any default locations for `/tmp/conf.jar`.

Reply:
@dimas-b thanks for checking. Let me try that later tonight. And yes, service config is still needed but not via JAR. This is currently getting populated by `.polarisServerConfig` in the values.yaml.

Reply:
Yes, but my point is that the location of the generated `conf.jar` should be in the server config too.

Reply:
Perhaps we can merge the `metaStoreManager.conf-file` property with the other server properties in the ConfigMap.

Reply:
so the location of `conf.jar` is in the server config. Here is the how u can set it:
```
type: eclipse-link
persistence-unit: polaris
conf-file: XXXXX
```
Here is the ref: https://github.com/polaris-catalog/polaris/blob/2aab551e53f6c343ea0bab46c443c2da11f0ed2b/extension/persistence/eclipselink/src/main/java/io/polaris/extension/persistence/impl/eclipselink/EclipseLinkPolarisMetaStoreManagerFactory.java#L35

Currently I am not setting validation on the config. I think all we needed here is list this option in the `values.yaml` then make it configurable on the k8s Deployment/Job. What do you think?

Reply:
From my POV the main idea of having helm charts is to simplify users' life when deploying Polaris to k8s. So, I think we should "tie loose ends" in the sense of making the user configure only what is absolutely required, for example CPU/mem limits, passwords to external systems, etc.

`persistence.xml` is a low-level config file. I do not think users need to deal with it directly. If Eclipselink requires having this file in a jar file, I think it would be preferable to make a (small) java tool (class) for generating `persistence.xml` from higher-level parameters as part of pod startup (not to bother with a separate init container).

I mean: we can make helm work with `persistence.xml` inside a k8s secret as a whole, but is it actually helpful? If I were to have to deploy Polaris I'd prefer to build a custom docker image with all the configs adjusted to my specific needs rather than managing an XML config in a secret.

Reply:
I do agreed the current approach for conf.jar is pretty messy due to the limitation of EclipseLink. However, only provide user to enter username/password/jdbc user will probably not be a prod graded (e.g. default num of connection pool for EclipseLink is 32 which is really useful for a catalog). However, I do agreed that is very details and low level. 

I won't think most people will want to make their own docker image (not only because there is no doc for ref to conf.jar and why that is needed) as those will be more complex for them IMO (however, as of now people will need to build their own image due to lack of driver dependencies). The default docker image is really just not prod graded if an end-user tried to use it for anything close to prod upgrade as a simple config change will mean a complete new docker image (e.g. adjust connection pool from 32 to 64).

Then for simplicity, i think we set the default to in-memory which achieved this purpose already?

Reply:
Starting with in-memory only helm chart SGTM :)

Reply:
Sounds good and that is already the case I think with the default value I set in the values.yaml. Unless you want to remove other things such as replication HPA etc to avoid confusion?

Reply:
Sounds good and that is already the case I think with the default value I set in the values.yaml. Unless you want to remove other things such as replication HPA etc to avoid confusion?

Reply:
I'm going to be away for some days... will review later.

Reply:
The change from `/tmp` to `/eclipselink-config` is completed and added a note in `values.yaml` regarding not to change it when using custom `persistence.xml` (by default, it will still be in-memory. I think this should be acceptable for now and hopefully later on once devs fix the EclipseLink issues we mentioned above, we can then come back optimize this part. What do you think?